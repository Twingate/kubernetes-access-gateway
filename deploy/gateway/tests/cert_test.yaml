suite: TLS Certificate
templates:
  - cert.yaml
set:
  tls:
    autoGenerated:
      enabled: true
      engine: cert-manager
tests:
  - it: should not render when `autoGenerated` is disabled
    set:
      tls:
        autoGenerated:
          enabled: false
    asserts:
      - hasDocuments:
          count: 0
  - it: should not render when `engine` is helm
    set:
      tls:
        autoGenerated:
          engine: helm
    asserts:
      - hasDocuments:
          count: 0
  - it: should not render when `existingSecret` is provided
    set:
      tls:
        existingSecret: "my-tls-secret"
    asserts:
      - hasDocuments:
          count: 0
  - it: should render
    set:
      twingate:
        resource:
          enabled: true
          extraAnnotations:
            resource.twingate.com/alias: my-alias.int
      tls:
        dnsNames:
          - foo.int
        ipAddresses:
          - 1.1.1.1
    asserts:
      - matchSnapshot: {}
  - it: should set `keyAlgorithm`
    set:

      tls:
        autoGenerated:
          certManager:
            keyAlgorithm: RSA
    asserts:
      - equal:
          path: spec.privateKey.algorithm
          value: RSA
  - it: should set `keySize`
    set:
      tls:
        autoGenerated:
          certManager:
            keySize: 2048
    asserts:
      - equal:
          path: spec.privateKey.size
          value: 2048
  - it: should set `duration`
    set:
      tls:
        autoGenerated:
          certManager:
            duration: 2160h
    asserts:
      - equal:
          path: spec.duration
          value: 2160h
  - it: should set `renewBefore`
    set:
      tls:
        autoGenerated:
          certManager:
            renewBefore: 360h
    asserts:
      - equal:
          path: spec.renewBefore
          value: 360h
