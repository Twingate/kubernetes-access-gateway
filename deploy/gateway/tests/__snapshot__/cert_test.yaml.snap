should render:
  1: |
    apiVersion: cert-manager.io/v1
    kind: Certificate
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: kubernetes-access-gateway
        app.kubernetes.io/version: 0.1.0
        helm.sh/chart: kubernetes-access-gateway-0.1.0
      name: RELEASE-NAME-kubernetes-access-gateway-crt
    spec:
      commonName: kubernetes.default.svc.cluster.local
      dnsNames:
        - kubernetes
        - kubernetes.default
        - kubernetes.default.svc
        - kubernetes.default.svc.cluster.local
        - my-alias.int
        - foo.int
        - 1.1.1.1
      duration: 2160h
      issuerRef:
        kind: Issuer
        name: RELEASE-NAME-kubernetes-access-gateway-ca-issuer
      privateKey:
        algorithm: RSA
        rotationPolicy: Always
        size: 2048
      renewBefore: 360h
      secretName: RELEASE-NAME-kubernetes-access-gateway-tls
      subject:
        organizations:
          - Twingate
